### PR TITLE
Update sort to have nulls at last

### LINKED ISSUE #17316: Sort "Recently Published" on Dashboard show unpublished articles on top
**Describe the bug**

Unpublished articles are shown on top while doing the sort "Recently Published" on the dashboard.

**To Reproduce**
- Create two articles
- Publish one of them
- Go to Dashboard
- Select "Recently Published" sort
- Unpublished articles will come first 

**Expected behavior**

It should show the published post in descending order and the unpublished posts at the bottom.

**Screenshots**

**Desktop (please complete the following information):**

- OS, version: Ubuntu 16.04
- Browser, version: Chrome Latest

### PR DESCRIPTION
**Pull request by Anshuman71**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Fix the sorting behavior to show Unpublished posts at the end.

## Related Tickets & Documents
Closes #17316 

## QA Instructions, Screenshots, Recordings

I don't know Ruby so please review the changes. I've done a lot of googling to update the syntax.

## [optional] What gif best describes this PR or how it makes you feel?

proud

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -364,7 +364,7 @@ def self.unique_url_error
         :created_at
       end
 
-    order(column => dir.to_sym)
+    order("#{column} #{dir}")
```
Comment (msarit) [inline app/models/article.rb:367]: Hello @Anshuman71, thank you for your PR.

I will defer to other reviewers, but your change on this line seems less readable to me. I would remove the reassignment on line `355` and then on line `367`: `order(column: dir.to_sym)`

What do you think?

**Review by msarit — CHANGES_REQUESTED** [review]

File: app/models/article.rb (hunk 1 above)
Comment (jeremyf) [inline app/models/article.rb:367]: Another concern I have is if this opens up SQL injection.  We'll need to investigate this approach.

**Review by jeremyf — CHANGES_REQUESTED** [review]

File: app/models/article.rb (hunk 1 above)
Comment (Anshuman71) [reply app/models/article.rb:367]: @jeremyf thanks for the feedback, I don't know Ruby at all I just looked up the correct answer. My goal was to highlight one possible solution.

Comment (Anshuman71) [reply app/models/article.rb:367]: @msarit  thanks for the feedback, I don't know Ruby at all I just looked up the correct answer. My goal was to highlight one possible solution.

**Comment by jeremyf** [comment]
I'm going to close this PR as there is still discussion around the approach for the desired behavior.

**Comment by jeremyf** [comment]
I went ahead and added another commit to this PR to rework our `Article.sorting` scope; the `.sorting` method is used in one location so adjusting the scope will help it better match @amywtlin's desired state: "I want the drafts filtered out in Recently published"

Further, we're building from the proposed work @Anshuman71 put forward, so I want to make sure that they get the "credit" for the PR.  Thank you!

**Comment by jeremyf** [comment]
@Anshuman71 would you be able to rebase / merge from main to update this code?  The breaking build we're seeing is resolved with https://github.com/forem/forem/pull/17390

**Comment by Anshuman71** [comment]
thanks, @jeremyf. I've merged the latest `main` into the branch. Let's see if it works out, also, thanks for recommending "credit". 😃

**Review by jeremyf — APPROVED** [review]

**Review by msarit — APPROVED** [review]
Amazing! 🚀 

This certainly mitigates SQL injection by placing the entire `order()` call within the case statements, instead of passing the `column` value from the case statements into `order()`. 🎉 

Thank you @Anshuman71 and @jeremyf!

**Pull request merged**